src/lang/modifyAst.ts

```diff
@@ -316,64 +358,34 @@ export function extrudeSketch({
     'PipeExpression'
   )
   if (err(_node2)) return _node2
-  const { node: pipeExpression } = _node2
-
-  const isInPipeExpression = pipeExpression.type === 'PipeExpression'
 
   const _node3 = getNodeFromPath<VariableDeclarator>(
     _node,
     pathToNode,
     'VariableDeclarator'
   )
   if (err(_node3)) return _node3
-  const { node: variableDeclarator, shallowPath: pathToDecleration } = _node3
+  const { node: variableDeclarator } = _node3
 
-  const sketchToExtrude = shouldPipe
-    ? createPipeSubstitution()
-    : createIdentifier(variableDeclarator.id.name)
-  const extrudeCall = createCallExpressionStdLibKw('extrude', sketchToExtrude, [
-    createLabeledArg('length', distance),
-  ])
+  const extrudeCall = createCallExpressionStdLibKw(
+    'extrude',
+    createIdentifier(variableDeclarator.id.name),
+    [createLabeledArg('length', distance)]
+  )
   // index of the 'length' arg above. If you reorder the labeled args above,
   // make sure to update this too.
   const argIndex = 0
 
-  if (shouldPipe) {
-    const pipeChain = createPipeExpression(
-      isInPipeExpression
-        ? [...pipeExpression.body, extrudeCall]
-        : [sketchExpression as any, extrudeCall]
-    )
-
-    variableDeclarator.init = pipeChain
-    const pathToExtrudeArg: PathToNode = [
-      ...pathToDecleration,
-      ['init', 'VariableDeclarator'],
-      ['body', ''],
-      [pipeChain.body.length - 1, 'index'],
-      ['arguments', 'CallExpressionKw'],
-      [argIndex, ARG_INDEX_FIELD],
-      ['arg', LABELED_ARG_FIELD],
-    ]
-
-    return {
-      modifiedAst: _node,
-      pathToNode,
-      pathToExtrudeArg,
-    }
-  }
-
   // We're not creating a pipe expression,
   // but rather a separate constant for the extrusion
   const name =
     extrudeName ?? findUniqueName(node, KCL_DEFAULT_CONSTANT_PREFIXES.EXTRUDE)
   const VariableDeclaration = createVariableDeclaration(name, extrudeCall)
 
-  const sketchIndexInPathToNode =
-    pathToDecleration.findIndex((a) => a[0] === 'body') + 1
-  const sketchIndexInBody = pathToDecleration[
-    sketchIndexInPathToNode
-  ][0] as number
+  const lastSketchNodePath =
+    orderedSketchNodePaths[orderedSketchNodePaths.length - 1]
+
+  const sketchIndexInBody = Number(lastSketchNodePath[1][0])
```

Comment:
Ah dang my edit flows PR was not orthogonal, I'm sorry I should have just diffed it

Reply:
Not sure I follow.